### PR TITLE
internal/util/url: don't require networking for empty schemes

### PR DESCRIPTION
**Pull request by arithx**
Right now Ignition will incorrectly report that it needs networking when
an empty scheme is given. Modify the `UrlNeedsNet` function to allow
both `data` & empty schemes.

A common case that could hit this is a config that writes a file with
empty contents.

### CODE REVIEW

File: internal/exec/stages/fetch_offline/fetch_offline_test.go [hunk 1]
```diff
@@ -64,6 +64,8 @@ func TestConfigNeedsNet(t *testing.T) {
 		},
 	}
 	assertNotNeedsNet(t, &cfg)
+	cfg.Storage.Files[0].Contents.Source = util.StrToPtr("")
```
Comment (jlebon) [inline internal/exec/stages/fetch_offline/fetch_offline_test.go:67]: Can we also add a test for `nil` here? (I.e. not specified). This tests `sourceNeedsNet`.

Comment (arithx) [reply internal/exec/stages/fetch_offline/fetch_offline_test.go:67]: Done.

**Comment by jlebon** [comment]
Blackbox tests flake in CoreOS CI. I restarted it. At some point, we should dig into what's going on exactly there.
